docs/advanced-topics/client-anticipation.md

```diff
@@ -0,0 +1,117 @@
+
+## StaleDataHandling
+
+On problem that an anticipation system needs to be able to solve is the problem of **stale data**. Stale data refers to updates from the server that represent actions that happened before your last request, and are actually going to be overwritten by that request.
+
+Let's return to the example from above of an object whose color gets changed, but let's expand it to include a second client who's also trying to change the color of the same object. If client 1 tries to change the object to blue, and then client 2 tries to change it to red, client 1 will see a delayed switch to blue, followed by a switch to red (which is fine because this is actually what happened). Client 2, however, will click the button to change it to red, then see it change to blue, followed by a change to red.
+
+<figure>
+<ImageSwitcher 
+lightImageSrc="/sequence_diagrams/Anticipation/ServerAuthoritativeMultiClient.png?text=LightMode"
+darkImageSrc="/sequence_diagrams/Anticipation/ServerAuthoritativeMultiClient_Dark.png?text=DarkMode"/>
+</figure>
+
+With client anticipation, we change the way this works: client 1 anticipates the change to blue so it happens immediately, and then later sees the object change to red, which, again, is fine. Client 2 also sees the object change to red immediately, but because a change to blue is already in progress, that overwrites client 2's anticipated value, causing it to flicker briefly to blue from client 1's request before changing back to red again from client 2's request.
+
+<figure>
+<ImageSwitcher 
+lightImageSrc="/sequence_diagrams/Anticipation/StaleDataNoPolicy.png?text=LightMode"
+darkImageSrc="/sequence_diagrams/Anticipation/StaleDataNoPolicy_Dark.png?text=DarkMode"/>
+</figure>
+
+To address this, Netcode for GameObjects's client anticipation includes a feature called `StaleDataHandling`. Stale data is determined based on assumptions about causation - it assumes that when you make an anticipation on the client side based on player input, that at the same time, you will be sending an RPC to the server requesting it to make the same change. It uses a continuously incrementing `AnticipationCounter` to track when the server has received and responded to the batch of requests that was sent on the same frame as the variable was anticipated. If an update for a variable arrives before the server has processed that message, the anticipation system regards that data as being "stale".
```

Comment:
Part of this feels like implementation details that could be put a special comment box? Just to make it easier to skim over when fast reading?